peekingduck/pipeline/nodes/augment/undistort.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Undistorts an image by removing `radial distortion
+<https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html>`_.
+"""
+
+
+from typing import Any, Dict
+
+from pathlib import Path
+import cv2
+import yaml
+import numpy as np
+
+from peekingduck.pipeline.nodes.abstract_node import AbstractNode
+
+
+class Node(AbstractNode):
+    """Undistorts an image by removing `radial distortion
+    <https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html>`_.
+    This may help to improve the performance of certain models.
+
+    Before using this node for the first time, please follow the tutorial in
+    :mod:`dabble.camera_calibration` to calculate the camera coefficients
+    of the camera you are using, and ensure that the file_path that the
+    coefficients are stored in is the same as the one specified in the configs.
+
+    The images below show an example of an image before and after undistortion
+
+    .. figure:: /assets/api/undistort.png
+        :width: 80 %
+
+        Before undistortion (left) and after undistortion (right)
+
+    Inputs:
+        |img_data|
+
+    Outputs:
+        |img_data|
+
+    Configs:
+        file_path (:obj:`str`):
+            **default = "PeekingDuck/data/camera_calibration_coeffs.yml"**. |br|
+            Path of the YML file containing calculated camera coefficients.
+    """
+
+    def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
+        super().__init__(config, node_path=__name__, **kwargs)
+
+        self.file_path = Path(self.file_path)  # type: ignore
+        # check if file_path has a ".yml" extension
+        if self.file_path.suffix != ".yml":
+            raise ValueError("Filepath must have a '.yml' extension.")
+        if not self.file_path.exists():
+            raise FileNotFoundError(
+                f"File {self.file_path} does not exist. "
+                "Please run the camera calibration again."
```

Comment:
Perhaps can mention the `dabble.camera_calibration` node here and link to the RTD page: https://peekingduck.readthedocs.io/en/stable/nodes/dabble.camera_calibration.html

For users who try undistortion without reading our docs first.